dids/src/test/kotlin/web5/sdk/dids/methods/dht/DhtTest.kt

```
@@ -0,0 +1,169 @@
+
+      val message = did.didDocument?.let { DidDht.toDnsPacket(it) }
+      assertNotNull(message)
+
+      val bep44Message = Dht.createBep44PutRequest(manager, kid, message)
+      assertNotNull(bep44Message)
+
+      val parsedMessage = Dht.parseBep44GetResponse(bep44Message)
+      assertNotNull(parsedMessage)
+
+      assertEquals(message.toString(), parsedMessage.toString())
+    }
+
+    @Test
+    fun `put and get a bep44 message to a pkarr relay`() {
+      val dht = Dht()
+      val manager = InMemoryKeyManager()
+      val did = DidDht.create(manager)
+
+      require(did.didDocument != null)
+
+      val kid = did.didDocument!!.verificationMethods?.first()?.publicKeyJwk?.get("kid")?.toString()
+      assertNotNull(kid)
+
+      val message = did.didDocument?.let { DidDht.toDnsPacket(it) }
+      assertNotNull(message)
+
+      val bep44Message = Dht.createBep44PutRequest(manager, kid, message)
+      assertNotNull(bep44Message)
+
+      assertDoesNotThrow { dht.pkarrPut(did.suffix(), bep44Message) }
+
+      // sleep 10 seconds to wait for propagation
+      Thread.sleep(10000)
+
+      val retrievedMessage = assertDoesNotThrow { dht.pkarrGet(did.suffix()) }
+      assertNotNull(retrievedMessage)
+    }
+
+    @Test
+    fun `bad pkarr put`() {
+      val dht = Dht()
+      val manager = InMemoryKeyManager()
+      val did = DidDht.create(manager)
+
+      require(did.didDocument != null)
+
+      val kid = did.didDocument!!.verificationMethods?.first()?.publicKeyJwk?.get("kid")?.toString()
+      assertNotNull(kid)
+
+      val message = did.didDocument?.let { DidDht.toDnsPacket(it) }
+      assertNotNull(message)
+
+      val bep44Message = Dht.createBep44PutRequest(manager, kid, message)
+      assertNotNull(bep44Message)
+
+      assertThrows<IllegalArgumentException> { dht.pkarrPut("bad", bep44Message) }
+    }
+
+    @Test
+    fun `bad pkarr get`() {
+      val dht = Dht()
+
+      assertThrows<IllegalArgumentException> { dht.pkarrGet("bad") }
```

Consider also asserting the message content.

---

what do you mean by this?

---

```suggestion
      assertThrows<IllegalArgumentException>("expected message from bad call") { dht.pkarrGet("bad") }
```

---

oh nice will update 😄

---

noting this doesn't actually assert the message

```/**
 * Example usage:
 * ```kotlin
 * val exception = assertThrows<IllegalArgumentException>("Should throw an Exception") {
 *     throw IllegalArgumentException("Talk to a duck")
 * }
 * assertEquals("Talk to a duck", exception.message)
 * ```
 * @see Assertions.assertThrows
 */
inline fun <reified T : Throwable> assertThrows(message: String, executable: () -> Unit): T =
    assertThrows({ message }, executable)```

---

TIL, thanks. The proper way to do so is https://github.com/TBD54566975/web5-kt/blob/234416a5e10872401af5f28e2a419c2d297d6540/dids/src/test/kotlin/web5/sdk/dids/methods/ion/DidIonTest.kt#L554